src/test/pybind/test_rbd.py

```diff
@@ -2391,6 +2391,24 @@ def test_site_name(self):
         self.rbd.mirror_site_name_set(rados, "")
         eq(rados.get_fsid(), self.rbd.mirror_site_name_get(rados))
 
+    def test_mirror_remote_namespace(self):
+        remote_namespace = "remote-ns"
+        # cannot set remote namespace for the default namespace
+        assert_raises(InvalidArgument, self.rbd.mirror_remote_namespace_set,
+                      ioctx, remote_namespace)
+        assert_raises(ImageNotFound, self.rbd.mirror_remote_namespace_get, ioctx)
+        self.rbd.namespace_create(ioctx, "ns1")
+        ioctx.set_namespace("ns1")
+        self.rbd.mirror_mode_set(ioctx, RBD_MIRROR_MODE_IMAGE)
+        # cannot set mirror remote namespace while mirroring enabled
+        assert_raises(InvalidArgument, self.rbd.mirror_remote_namespace_set,
+                      ioctx, remote_namespace)
+        self.rbd.mirror_mode_set(ioctx, RBD_MIRROR_MODE_DISABLED)
+        self.rbd.mirror_remote_namespace_set(ioctx, remote_namespace)
+        eq(remote_namespace, self.rbd.mirror_remote_namespace_get(ioctx))
+        ioctx.set_namespace("")
```

Comment:
Let's add a "cannot set remote namespace to the default namespace" assert here before switching out of ns1.